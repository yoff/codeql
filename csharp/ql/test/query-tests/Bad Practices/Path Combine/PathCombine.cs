using System.IO;

class EmptyCatchBlock
{
    void bad()
    {
        Path.Combine(@"C:\Users", @"C:\Program Files");
    }

    void good()
    {
        Path.Join(@"C:\Users", @"C:\Program Files");
    }
}
