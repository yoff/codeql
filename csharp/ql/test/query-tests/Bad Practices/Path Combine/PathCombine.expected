| PathCombine.cs:7:9:7:54 | call to method Combine | Path.Combine may silently discard its initial arguments if the latter are absolute paths. Use Path.Join to consistently join them. |
